packages/salesforcedx-vscode-core/src/commands/templates/forceFunctionCreate.ts

```diff
@@ -47,33 +50,42 @@ export class ForceFunctionCreateExecutor extends BaseTemplateCommand {
       .build();
   }
 
-  public runPostCommandTasks(targetDir: string) {
-    if (sfdxCoreSettings.getFunctionsPullDependencies()) {
-      vscode.window.withProgress(
-        {
-          location: vscode.ProgressLocation.Window,
-          title: nls.localize(
-            'force_function_install_npm_dependencies_progress'
-          ),
-          cancellable: true
-        },
-        () => {
-          return new Promise((resolve, reject) => {
-            cp.exec('npm install', { cwd: targetDir }, err => {
-              if (err) {
-                notificationService.showWarningMessage(
-                  nls.localize(
-                    'force_function_install_npm_dependencies_error',
-                    err.message
-                  )
-                );
-                reject(err);
-              }
-              resolve();
-            });
-          });
-        }
-      );
+  public runPostCommandTasks(data: FunctionInfo) {
+    const language = data.language;
+    if (language === LANGUAGE_JAVA) {
+      const pathToSource = this.getPathToSource(data.outputdir, data.fileName);
+      const targetDir = path.join(path.dirname(pathToSource), '..', '..', '..', '..', '..');
+      return new Promise((resolve, reject) => {
+        cp.exec('mvn install', { cwd: path.join(targetDir) }, err => {
+          if (err) {
+            notificationService.showWarningMessage(
+              nls.localize(
+                'force_function_install_mvn_dependencies_error',
+                err.message
+              )
+            );
+            reject(err);
+          }
+          resolve();
+        });
```

Comment:
Is it possible to stream the output to the output channel? I couldn't tell if anything was happening until it finally failed. I did have `mvn` installed and it wasn't until I switched to the function directory and ran it until i knew what the error was. I'm also not sure when the dependencies actually finish installing

Reply:
One of the acceptance criteria Stephanie wanted in this story was to "Remove all the user notifications we show at the time of dependency installation." So we removed all the notifications and make the dependencies installation as silent as possible.